index.js

```diff
@@ -37,76 +38,128 @@ function send(payload) {
-          name: ${process.env.KUBERNETES_CONTEXT}
-        current-context: ${process.env.KUBERNETES_CONTEXT}
-        kind: Config
-        preferences: {}
-        users:
-        - name: alamo-${process.env.KUBERNETES_CONTEXT}-admin
-          user:
-            client-certificate-data: ${(Buffer.from(cert['admin-crt'], 'utf8')).toString('base64')}
-            client-key-data: ${(Buffer.from(cert['admin-key'], 'utf8')).toString('base64')}`)
-    } else {
-      let cert = (await vc.read(process.env.KUBERNETES_CERT_SECRET)).data
-      let token = (await vc.read(process.env.KUBERNETES_TOKEN_SECRET)).data
-      assert.ok(cert['ca-crt'], 'The ca-crt was not found within the vault kubernetes secret')
-      kc.loadFromString(`
-        apiVersion: ${process.env.KUBERNETES_API_VERSION}
-        clusters:
-        - cluster:
-            insecure-skip-tls-verify: 'true'
-            certificate-authority-data: ${(Buffer.from(cert['ca-crt'], 'utf8')).toString('base64')}
-            server: https://${process.env.KUBERNETES_API_SERVER}
-          name: alamo-${process.env.KUBERNETES_CONTEXT}-cluster
-        contexts:
-        - context:
-            cluster: alamo-${process.env.KUBERNETES_CONTEXT}-cluster
-            namespace: kube-system
-            user: alamo-${process.env.KUBERNETES_CONTEXT}-admin
-          name: ${process.env.KUBERNETES_CONTEXT}
-        current-context: ${process.env.KUBERNETES_CONTEXT}
-        kind: Config
-        preferences: {}
-        users:
-        - name: alamo-${process.env.KUBERNETES_CONTEXT}-admin
-          user:
-            token: ${token.token}`)
+
+    if(!process.env.VAULT_ADDR.startsWith('http')) {
+      process.env.VAULT_ADDR = 'https://' + process.env.VAULT_ADDR
+    }
+
+    if(!process.env.KUBERNETES_API_SERVER.startsWith('http')) {
+      process.env.KUBERNETES_API_SERVER = 'https://' + process.env.KUBERNETES_API_SERVER
     }
+    
+    let vc = vault({apiVersion:'v1', endpoint:process.env.VAULT_ADDR, token:process.env.VAULT_TOKEN})
+
+    const token = (await vc.read(process.env.KUBERNETES_TOKEN_SECRET)).data
+
+    const cluster = {
+      name: `akkeris-${process.env.KUBERNETES_CONTEXT}-cluster`,
+      server: `${process.env.KUBERNETES_API_SERVER}`,
+    };
+
+    const user = {
+      name: `akkeris-${process.env.KUBERNETES_CONTEXT}-admin`,
+      token: token.token,
+    };
+    
+    const context = {
+      name: process.env.KUBERNETES_CONTEXT,
+      cluster: cluster.name,
+      user: user.name,
+    };
+
+    kc.loadFromOptions({
+      clusters: [cluster],
+      users: [user],
+      contexts: [context],
+      currentContext: context.name,
+    });
+
+    await checkPermissions();
+  } catch (err) {
+    console.log('\nVault loading failed with the following message:\n', err.message);
+    return false;
   }
-  if(process.env.KUBERNETES_CONTEXT) {
-    kc.setCurrentContext(process.env.KUBERNETES_CONTEXT)
+  return true;
+}
+
+async function connect_kube() {
```

Comment:
Minor but everything else is camelCase and this is kebab_case

Reply:
That wasn't me ;) lemme fix it real quick though